App/features/group/rename-group/sagas.ts

```diff
@@ -0,0 +1,26 @@
+import { ActionType, getType } from 'typesafe-actions'
+import { call, put, takeEvery, all } from 'redux-saga/effects'
+import Textile from '@textile/react-native-sdk'
+import { renameGroup } from './actions'
+import PhotoViewingActions from '../../../Redux/PhotoViewingRedux'
+
+export function *handleRenameGroupRequest(action: ActionType<typeof renameGroup.request>) {
+  const { threadId, name } = action.payload
+  try {
+    yield call(Textile.threads.rename, threadId, name)
+    yield put(renameGroup.success({ threadId }))
+    yield put(PhotoViewingActions.refreshThreadRequest(threadId))
+  } catch (error) {
+    yield put(renameGroup.failure({ threadId, error }))
+  }
+}
+
+export function *watchForRenameGroupRequests() {
+  yield takeEvery(getType(renameGroup.request), handleRenameGroupRequest)
```

Comment:
Been meaning to get rid of the pattern of having this "watch" function. Just put `takeEvery(getType(renameGroup.request), handleRenameGroupRequest)` directly into the `yield all([])` below.